### DISCUSSION TITLE
Question on the new App Router and dynamic routes...

### DISCUSSION THREAD

**Discussion by matthewgisonno** [vercel/next.js#48894] [App Router]
I'm reviewing the documentation here: https://beta.nextjs.org/docs/api-reference/generate-static-params#multiple-dynamic-segments

It shows a multi-part dynamic route of `app/products/[category]/[product].tsx` ... I can't seem to get the dynamic .tsx portion working.  If I use a dynamic folder of say `[category]` which contains a `page.tsx` it works fine.  But if I create a page of `[category].tsx` I cannot seem to get it working, returns a 404.

My work-around for now is to just turn the .tsx file into a folder that contains a `page.tsx` ... So `/[category].tsx` becomes `/[category]/page.tsx` ... Not ideal, but works.  Am I mis-reading the documentation?

It also lists dynamic page routes here: https://beta.nextjs.org/docs/api-reference/generate-static-params#single-dynamic-segment to which I put together a codesandbox around.

At first I thought it might be some wonky setup with my project, so I tried it out on codesandbox and same issue... Here is the codesandbox url: https://codesandbox.io/p/sandbox/clever-tdd-2byrx9

In the codesandbox, when trying to navigate to the dynamic route of `/product/1` it returns a 404.

**Comment by angeloashmore** [comment]
Creating a directory for `[category]` is correct, as evidenced by the `app/products/[category]/[product].tsx` file path in the docs.

In order for a page to appear in App Router, it must be registered using a `page.tsx` file. `[id].tsx` is not a valid page, but `[id]/page.tsx` is.
